tuning/tune.py

```diff
@@ -633,6 +638,63 @@ def get_default_output_dir():
 
     return "tuning_" + datetime.now().strftime("%Y_%m_%d_%H_%M")
 
+def parse_mlir(mlir_text):
+    mlir_module = None
+    with ireec.ir.Context() as context:
+        try:
+            mlir_module = ireec.ir.Module.parse(mlir_text)
+            tune_logger.info("MLIR parsing successful!")
+        except ireec.ir.MLIRError as e:
+            tune_logger.error(f"Error parsing MLIR: {e}")
+            raise RuntimeError(f"Error parsing MLIR: {e}")
+        
+    return mlir_module
+
+class OpWalkResult:
+    def __init__(self):
+        self.wasInterrupted = False
+        self.isConv = False
+        self.isMatmul = False
+        self.isContraction = False
+        self.isBatchMatmul = False
+
+    def set_interrupt(self):
+        self.wasInterrupted = True
+
+    def set_conv(self):
+        self.isConv = True
+
+    def set_matmul(self):
+        self.isMatmul = True
+
+    def set_contraction(self):
+        self.isContraction = True
+
+    def set_batch_matmul(self):
+        self.isBatchMatmul = True
+
+def walk_callback_detect_type(op: ireec._mlir_libs._mlir.ir.Operation, walk_result: OpWalkResult) -> ireec._mlir_libs._mlir.ir.WalkResult:
```

Comment:
Should we have a top-level definition for `ireec._mlir_libs._mlir` to avoid repeating these long paths?